core/src/main/scala/org/apache/spark/status/api/v1/api.scala

```diff
@@ -89,6 +89,14 @@ class ExecutorStageSummary private[spark](
     val peakMemoryMetrics: Option[ExecutorMetrics],
     val isExcludedForStage: Boolean)
 
+class SpeculationStageSummary private[spark](
```

Comment:
ditto.

Reply:
Do you think we don't need this anymore given the REST API is not required anymore?

Reply:
I think this class is still needed.
How about `SpeculationSummary` instead of `SpeculationStageSummary`?
Speculation is not about stages but tasks.

Reply:
Speculation summary can be at different level right like job, app, stage etc. which is why I named this `SpeculationStageSummary`.

Reply:
> Speculation summary can be at different level right like job, app, stage etc.

What do you mean? Speculation is about task.
http://spark.apache.org/docs/latest/configuration.html#scheduling

Reply:
I understand speculation is about a task. What I mean here is we can have the summary of speculated tasks aggregated for stage level or job or app level which is why I named the class as `SpeculationStageSummary` so that it will be clear that it is specific to the stage. Hope that clarifies it.

Reply:
O.K. We already have `ExecutorStageSummary` which means aggregated information for a stage. So, the name `SpeculationStageSummary` can be acceptable.